jax/_src/scipy/stats/_core.py

```diff
@@ -203,13 +203,28 @@ def sem(a: ArrayLike, axis: int | None = 0, ddof: int = 1, nan_policy: str = "pr
     array
   """
   b, = promote_args_inexact("sem", a)
+
+  input_shape = b.shape
+  if keepdims:
+    if axis is None:
+      output_shape = tuple(1 for i in input_shape)
+    else:
+      output_shape = tuple(1 if i == axis else s for i, s in enumerate(input_shape))
+  else:
+    if axis is None:
+      output_shape = ()
+    else:
+      output_shape = tuple(s for i, s in enumerate(input_shape) if i != axis)
```

Comment:
better I think would be to pass `keepdims` to `std`, `nanstd`, and `sum` below rather than adding all this extra shape logic here. What do you think?

Reply:
Yes, agree with you. But when I use it, `scipy_stats_test.py` fails the some tests.

Reply:
Then I'd figure out why it's failing and fix it 😁

Reply:
It fails for `axis=None` cases, especially.

Reply:
If you'd like me to help debug the failures and suggest solutions, then you could push what you have to this branch so I can see it.

Reply:
The current changes with extra shape logic passes all the tests. Even if I test with the following command.
```
JAX_NUM_GENERATED_CASES=50 pytest -n auto tests/scipy_stats_test.py -k testSEM
```

If I try passing `keepdims` to `std`, `nanstd`, and `sum`, some tests are failing.

I am pushing the commit with the changes suggested by you here. Please help me debug the failures and suggest solution. Thanks!